src/components/StartScriptGenerator.tsx

```diff
@@ -0,0 +1,217 @@
+import React, { useEffect, useState, useRef } from 'react';
+import '@site/src/css/start-script-generator.css';
+
+const markerPoints = [4, 8, 12, 16, 20];
+
+type FlagType = {
+    label: string;
+    value: string;
+    description: string;
+};
+
+const WINDOWS_AUTO_RESTART = ":start\n%%CONTENT%%\n\necho Server restarting...\necho Press CTRL + C to stop.\ngoto :start";
+const LINUX_AUTO_RESTART = "while [ true ]; do\n    %%CONTENT%%\n    echo Server restarting...\n    echo Press CTRL + C to stop.\ndone";
+
+const FLAGS: { [key: string]: FlagType } = {
+    AIKARS: {
+        label: 'Aikar\'s',
+        value: '-XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 -XX:InitiatingHeapOccupancyPercent=15 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:+PerfDisableSharedMem -XX:MaxTenuringThreshold=1 -Dusing.aikars.flags=https://mcflags.emc.gs -Daikars.new.flags=true -XX:G1NewSizePercent=30 -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=8M -XX:G1ReservePercent=20',
+        description: 'Optimized Minecraft flags by Aikar for better server performance.',
+    },
+    NONE: { label: 'None', value: '', description: 'No additional flags.' },
+    VELOCITY: {
+        label: 'Velocity',
+        value: '-XX:+UseG1GC -XX:G1HeapRegionSize=4M -XX:+UnlockExperimentalVMOptions -XX:+ParallelRefProcEnabled -XX:+AlwaysPreTouch -XX:MaxInlineLevel=15',
+        description: 'Flags recommended for use with the Velocity proxy server.',
+    },
+};
+
+const isServerSide = typeof document === 'undefined';
+
+const generateStartCommand = (memory: number, selectedFlag: FlagType, filename: string, guiEnabled: boolean, autoRestartEnabled: boolean, platform: string) => {
+    setTimeout(resizeOutput, 0);
+    let content = "";
+    const command = `java -Xmx${memory * 1024}M -Xms${memory * 1024}M ${selectedFlag.value} -jar ${filename} ${guiEnabled ? '' : '--nogui'}`;
```

Comment:
This doesn't properly handle the `NONE` flag type by having 2 empty spaces
![firefox_mIJDb3vcqI](https://github.com/PaperMC/docs/assets/40437205/b45ea788-94b2-41f4-aa20-128975edcd3c)